driver-redpanda/src/main/java/io/openmessaging/benchmark/driver/redpanda/tx/RedpandaBenchmarkDriver.java

```diff
@@ -0,0 +1,213 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class RedpandaBenchmarkDriver implements BenchmarkDriver {
+    private static final Logger log = LoggerFactory.getLogger(RedpandaBenchmarkDriver.class);
+    private static final ObjectMapper mapper = new ObjectMapper(new YAMLFactory())
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private static int INIT_RETRIES = 5;
+    private final AtomicInteger producerId = new AtomicInteger(0);
+    private Config config;
+    private List<BenchmarkProducer> producers = Collections.synchronizedList(new ArrayList<>());
+    private List<BenchmarkConsumer> consumers = Collections.synchronizedList(new ArrayList<>());
+    private Properties topicProperties;
+    private Properties producerProperties;
+    private Properties consumerProperties;
+    private AdminClient admin;
+
+    public static void putUuid(ByteBuffer bb, int offset, UUID uuid) {
+        bb.putLong(offset, uuid.getMostSignificantBits());
+        bb.putLong(offset + 8, uuid.getLeastSignificantBits());
+    }
+
+    public static UUID getUuid(ByteBuffer bb, int offset) {
+        bb.position(offset);
+        long high = bb.getLong();
+        long low = bb.getLong();
+        return new UUID(high, low);
+    }
+
+    @Override
+    public void initialize(File configurationFile, StatsLogger statsLogger) throws IOException {
+        config = mapper.readValue(configurationFile, Config.class);
+
+        Properties commonProperties = new Properties();
+        commonProperties.load(new StringReader(config.commonConfig));
+
+        producerProperties = new Properties();
+        commonProperties.forEach((key, value) -> producerProperties.put(key, value));
+        producerProperties.load(new StringReader(config.producerConfig));
+        producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+
+        consumerProperties = new Properties();
+        commonProperties.forEach((key, value) -> consumerProperties.put(key, value));
+        consumerProperties.load(new StringReader(config.consumerConfig));
+        consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+
+        topicProperties = new Properties();
+        topicProperties.load(new StringReader(config.topicConfig));
+
+        admin = AdminClient.create(commonProperties);
+
+        if (config.reset) {
+            int retry = 0;
```

Comment:
I assume this code is duplicated from somewhere. Would be good if we could extract it as I know this is has been changed in the past.

Reply:
Think there is a few cases below related to topic creation as well.